homeassistant/components/supla/light.py

```diff
@@ -0,0 +1,175 @@
+) -> None:
+    """Set up the Supla lights."""
+    if discovery_info is None:
+        return
+
+    _LOGGER.debug("Discovery: %s", pformat(discovery_info))
+
+    entities: list[LightEntity] = []
+    for device in discovery_info.values():
+        device_name = device["function_name"]
+        server_name = device["server_name"]
+
+        if device_name == SUPLA_DIMMER:
+            entities.append(
+                SuplaDimmerEntity(
+                    device,
+                    hass.data[DOMAIN][SUPLA_SERVERS][server_name],
+                    hass.data[DOMAIN][SUPLA_COORDINATORS][server_name],
+                )
+            )
+
+        elif device_name == SUPLA_RGBLIGHTING:
+            entities.append(
+                SuplaRGBLightingEntity(
+                    device,
+                    hass.data[DOMAIN][SUPLA_SERVERS][server_name],
+                    hass.data[DOMAIN][SUPLA_COORDINATORS][server_name],
+                )
+            )
+
+    async_add_entities(entities)
+
+
+class SuplaDimmerEntity(SuplaEntity, LightEntity):
+    """Representation of a Supla dimmable light."""
+
+    _attr_color_mode = ColorMode.BRIGHTNESS
+    _attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the switch."""
+        if ATTR_BRIGHTNESS in kwargs:
+            # Convert from Home Assistant brightness (0-255) to (0-100)
+            brightness = min(100, ceil(kwargs[ATTR_BRIGHTNESS] * 100 / 255.0))
+            await self.async_action("SET_RGBW_PARAMETERS", brightness=brightness)
+            return
+        await self.async_action("TURN_ON")
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the switch."""
+        await self.async_action("TURN_OFF")
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if switch is on."""
+        if state := self.channel_data.get("state"):
+            return state["on"]
+        return False
+
+    @property
+    def brightness(self) -> int | None:
+        """Return the brightness of the light."""
+        state = self.channel_data.get("state")
+        if not state or "brightness" not in state:
+            return None
+        brightness = state.get("brightness")
+        # Convert from (0-100) to Home Assistant brightness (0-255)
+        return min(255, ceil(brightness * 255 / 100.0))
+
+
+class SuplaRGBLightingEntity(SuplaEntity, LightEntity):
+    """Representation of a Supla RGB light."""
+
+    _attr_color_mode = ColorMode.HS
+    _attr_supported_color_modes = {ColorMode.HS}
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the switch."""
+        brightness = kwargs.get(ATTR_BRIGHTNESS) or self.brightness
+        hs_color = kwargs.get(ATTR_HS_COLOR) or self.hs_color
+        if not self.is_on and ATTR_HS_COLOR in kwargs and ATTR_BRIGHTNESS not in kwargs:
+            # When the light is off, brightness state is set to 0.
```

Comment:
What is "brightness state", do you mean `color_brightness` which is extracted from the state below?

Reply:
I think this refers to HA state entity, and its `brightness` property. (It's been a while, sorry for the uncertainty.)